policy/release/olm_test.rego

```diff
@@ -373,3 +373,34 @@ test_unmapped_references_none_found if {
 		with input.image.files as {"manifests/csv.yaml": manifest}
 		with input.image.config.Labels as {olm.manifestv1: "manifests/"}
 }
+
+test_allowed_registries if {
+	# This should pass since registry.io is a member of allowed_registry_prefixes
+	lib.assert_empty(olm.deny) with data.rule_data.pipeline_intention as "release"
+		with data.rule_data.allowed_registry_prefixes as ["registry.io", "registry.redhat.io"]
+		with input.image.config.Labels as {olm.manifestv1: "manifests/"}
+		with input.image.files as {"manifests/csv.yaml": manifest}
+}
+
+test_unallowed_registries if {
+	expected := {
+		{
+			"code": "olm.allowed_registries",
+			# regal ignore:line-length
+			"msg": "The \"registry.io/repository/image@sha256:cafe\" CSV image reference is not from an allowed registry.",
+			"term": "registry.io/repository/image@sha256:cafe",
+		},
+		{
+			"code": "olm.allowed_registries",
+			# regal ignore:line-length
+			"msg": "The \"registry.io/repository/image2@sha256:tea\" CSV image reference is not from an allowed registry.",
+			"term": "registry.io/repository/image2@sha256:tea",
+		},
+	}
+
+	# This expects failure as registry.io is not a member of allowed_registry_prefixes
+	lib.assert_equal_results(olm.deny, expected) with data.rule_data.pipeline_intention as "release"
+		with data.rule_data.allowed_registry_prefixes as ["registry.access.redhat.com", "registry.redhat.io"]
+		with input.image.config.Labels as {olm.manifestv1: "manifests/"}
+		with input.image.files as {"manifests/csv.yaml": manifest}
```

Comment:
Did you mean to create/use a `manifest` that includes references to the non-allowed registry `registry.io`?

Reply:
Yes, I wanted `olm.allowed_registries` to detect a non-allowed registry in this test, so I re-used the same manifest and excluded `registry.io` from `allowed_registry_prefixes`.

Reply:
Should I do something a bit different here @lcarva?